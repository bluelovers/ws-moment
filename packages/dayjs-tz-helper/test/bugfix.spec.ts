//@noUnusedParameters:false
/// <reference types="jest" />
/// <reference types="node" />
/// <reference types="expect" />

import { basename, extname } from 'path';
import tzDayjsSafeParse from '../src/index';
// @ts-ignore
import dayjs from 'dayjs';
// @ts-ignore
import utc from 'dayjs/plugin/utc';
// @ts-ignore
import timezone from 'dayjs/plugin/timezone';

export const TZ_TIME_ZONE = "Asia/Taipei" as const

beforeAll(async () =>
{
	dayjs.extend(utc);
	dayjs.extend(timezone);

	dayjs.tz.setDefault(TZ_TIME_ZONE);
});

describe(basename(__filename, extname(__filename)), () =>
{

	test.skip(`dummy`, () => {});

	[
		'2023-03-31T04:00:00.000Z',
		'2023-03-31T12:00:00+08:00',
		'2023-03-31T04:00:00.800Z',
		'2023-03-31T04:00:00+00:00',
		'2023-03-31T12:00:00',
	].forEach(input => {

		test(input, () =>
		{

			let actual = tzDayjsSafeParse(input);
			let expected;

			expect({
				format: actual.tz('GMT').format(),
				format2: actual.tz(TZ_TIME_ZONE).format(),
				toISOString: actual.toISOString(),
				toJSON: actual.toJSON(),
				unix: actual.unix(),
			}).toMatchSnapshot({
				format: '2023-03-31T04:00:00Z',
				format2: '2023-03-31T12:00:00+08:00',
				unix: 1680235200,
			});

		});

	});

})
