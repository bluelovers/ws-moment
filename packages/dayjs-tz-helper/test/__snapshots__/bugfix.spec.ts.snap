// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bugfix.spec 2023-03-31T04:00:00+00:00 1`] = `
{
  "format": "2023-03-31T04:00:00Z",
  "format2": "2023-03-31T12:00:00+08:00",
  "toISOString": "2023-03-31T04:00:00.000Z",
  "toJSON": "2023-03-31T04:00:00.000Z",
  "unix": 1680235200,
}
`;

exports[`bugfix.spec 2023-03-31T04:00:00.000Z 1`] = `
{
  "format": "2023-03-31T04:00:00Z",
  "format2": "2023-03-31T12:00:00+08:00",
  "toISOString": "2023-03-31T04:00:00.000Z",
  "toJSON": "2023-03-31T04:00:00.000Z",
  "unix": 1680235200,
}
`;

exports[`bugfix.spec 2023-03-31T04:00:00.800Z 1`] = `
{
  "format": "2023-03-31T04:00:00Z",
  "format2": "2023-03-31T12:00:00+08:00",
  "toISOString": "2023-03-31T04:00:00.800Z",
  "toJSON": "2023-03-31T04:00:00.800Z",
  "unix": 1680235200,
}
`;

exports[`bugfix.spec 2023-03-31T12:00:00 1`] = `
{
  "format": "2023-03-31T04:00:00Z",
  "format2": "2023-03-31T12:00:00+08:00",
  "toISOString": "2023-03-31T04:00:00.000Z",
  "toJSON": "2023-03-31T04:00:00.000Z",
  "unix": 1680235200,
}
`;

exports[`bugfix.spec 2023-03-31T12:00:00+08:00 1`] = `
{
  "format": "2023-03-31T04:00:00Z",
  "format2": "2023-03-31T12:00:00+08:00",
  "toISOString": "2023-03-31T04:00:00.000Z",
  "toJSON": "2023-03-31T04:00:00.000Z",
  "unix": 1680235200,
}
`;
